src/samplechains.jl

```diff
@@ -0,0 +1,129 @@
+using .SampleChains: SampleChains
+using .SampleChains.TupleVectors: TupleVector
+
+# TODO: supported samples with a nested structure
+
+function namedtuple_of_arrays(x::TupleVector{<:NamedTuple{K}}) where {K}
+    return NamedTuple(k => flatten(getproperty(x, k)) for k in K)
+end
+function namedtuple_of_arrays(chain::SampleChains.AbstractChain)
+    return namedtuple_of_arrays(SampleChains.samples(chain))
+end
+function namedtuple_of_arrays(multichain::SampleChains.MultiChain)
+    chains = SampleChains.getchains(multichain)
+    return namedtuple_of_arrays(map(namedtuple_of_arrays, chains))
+end
+
+_maybe_multichain(x) = x
+_maybe_multichain(x::SampleChains.MultiChain) = x
+_maybe_multichain(x::SampleChains.AbstractChain) = SampleChains.MultiChain(x)
+function _maybe_multichain(x::AbstractVector{<:SampleChains.AbstractChain})
+    return SampleChains.MultiChain(x)
+end
+function _maybe_multichain(x::Tuple{Vararg{<:SampleChains.AbstractChain}})
+    return SampleChains.MultiChain(x...)
+end
+
+# info(::AbstractChain) is only required to return an AbstractVector, which is not enough
+# information for us to convert it
+# see https://github.com/arviz-devs/ArviZ.jl/issues/124
+_samplechains_info(::SampleChains.AbstractChain) = nothing
+function _samplechains_info(multichain::SampleChains.MultiChain)
+    stats = map(_samplechains_info, SampleChains.getchains(multichain))
+    all(isnothing, stats) && return nothing
+    return namedtuple_of_arrays(stats)
+end
+
+@require SampleChainsDynamicHMC = "6d9fd711-e8b2-4778-9c70-c1dfb499d4c4" begin
+    using .SampleChainsDynamicHMC: SampleChainsDynamicHMC
+
+    function _samplechains_info(chain::SampleChainsDynamicHMC.DynamicHMCChain)
+        info = SampleChains.info(chain)
+        termination = info.termination
+        tree_stats = (
+            lp=info.π,
```

Comment:
On further inspection, `π` is [documented](https://github.com/tpapp/DynamicHMC.jl/blob/v3.1.0/src/NUTS.jl#L207-L208) as the log density of the proposed phasepoint, i.e. the energy. It's not clear if the value of the log density for that sample is returned anywhere.

```suggestion
            energy=info.π,
```